app/controllers/api/restful_controller.rb

```diff
@@ -18,7 +20,8 @@ def destroy_action
   private
 
   def current_user
-    super || LoggedOutUser.new
+    token_user = User.find(doorkeeper_token.resource_owner_id) if doorkeeper_token
```

Comment:
Not quite sure I follow; I think this method is the only thing we use anywhere in the API to refer to the current user, so modifying this method should cover the whole of the API.

Might be nice to have some testing around this?